keymanapp/keyman
# fix(android/engine): Test fileVersion during package install

Addresses #6254 for 15.0 beta. Will need to 🍒 pick to stable.14.0 and master to fix the issue.

This updates PackageActivity() to check kmp.json that the minimum keyboard version (system.fileVersion) is supported by the current Keyman version.

### Notes on test keyboards
The test keyboard "the_99.kmp" is a standard keyboard created from template where fileVersion is manually edited to "99.0". 
Keyman Developer currently (as of 15.0.210 beta) doesn't allow `store(&VERSION)` to define version greater than '15.0'.

Also updated full_caps_3620_3621 to use `store(&VERSION) '15.0'`

## User Testing

Setup
1. On an Android device/emulator, load the PR build of Keyman for Android
2. Also upload the [test_99.kmp](https://github.com/keymanapp/keyman/raw/b213c3b8291221746d96013769308430d7716f04/common/test/keyboards/the_99/the_99.kmp) keyboard to the device's "Download" folder.


* **TEST_FILEVERSION_CHECKED**
1. Launch Keyman
4. Dismiss the "Get Started" menu
5. From the "Settings" menu --> "Install Keyboard or Dictionary" --> Install from local file 
6. Browse to the Downloads folder where the test_99.kmp file was uploaded.
7. Install the test_99.kmp keyboard and verify the installation fails
Screenshot:
 
![unsupported](https://user-images.githubusercontent.com/7358010/157601788-fbe15db0-9b01-4195-b42f-da9c9ba1a508.png)

8. From the "Settings" menu, install the khmer_angkor keyboard from keyman.com
9. Verify the khmer_angkor keyboard downloads and installs.
